transformer_engine/pytorch/tensor/utils.py

```diff
@@ -305,4 +305,11 @@ def _cast_master_weights_to_fp8_current_scaling(params, group, use_fsdp_shard_mo
             amax=torch.Tensor(),
             fp8_dtype=model_weight._fp8_dtype,
         )
+        if use_fsdp_shard_model_weights and not isinstance(model_weight_fragment, Float8Tensor):
```

Comment:
Is this just to handle the case when the FP8 weight is internal? I.e. Float8TensorBase?